moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_arm_data.h

```diff
@@ -101,6 +102,9 @@ struct JogArmShared : public std::mutex
 
   // Stop jog loop threads - threads are not stopped by default
   std::atomic<bool> stop_requested{ false };
+
+  // Use a lock-free queue to update filter coefficients from other threads
+  boost::lockfree::queue<double> filter_coefficient_queue;
```

Comment:
Suggest using `spsc_queue` (https://www.boost.org/doc/libs/1_72_0/doc/html/boost/lockfree/spsc_queue.html) if this is only published to by one thread and read from another for a minor performance benefit over queue.

Reply:
The idea is to have maybe typically one thread adjusting the filter coefficient in a thread-safe manner? Wouldn't it be much easier to just modify the code a bit to make `JogCalcs::updateFilterCoeff` thread-safe and allow threads to use that directly? If you aren't consuming the filter coefficients at the same rate as you are producing them or faster, the queue size will explode unless you set a fixed size for it.